### PR TITLE
Config Helper Integration

### PR DESCRIPTION
**Pull request by sksinha2410**
### Description
- Added separate files for Constants.
- YAML Configs in place : 
          - `colors.yaml`
          - `fonts.yaml`
          - `FontSize.yaml` 
          - `Alert.yaml`
          - `ErrorMessage.yaml`
          - `TimeFormate.yaml` 
          - `Design.yaml`
- Organised constants based on their function.
- Created `ConfigHelper.js` that helps to use these constants in .tsx files

### Example Usage

 To use the constant:  **import 'constants' from './src/UI-Constants/Constant.json';**

 Now you can simply use **'{constants.NETWORK_ERROR}'** to use the constant where **NETWORK_ERROR** will be replaced by the **name of the constant** you want to use

### CODE REVIEW

**Comment by Jha2508** [comment]
@sksinha2410 can you try yaml files for the purpose ? will be much more optimized in cases for app size and memory solutions as these constants will be there throughout the lifecycle of the app resulting in bulking up the app size . Try using `js-yaml` and `@types/js-yaml`

File: src/Utilities/ConfigHelper.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        return null;
+    }
+};
+
+export const DefinedColors = await loadConfig('./src/UI-Constants/Colors.yaml');
+export const DefinedDesign = await loadConfig('./src/UI-Constants/Designs.yaml');
+export const DefinedErrors = await loadConfig('./src/UI-Constants/Errors.yaml');
+export const FontSizeWeightFamily = await loadConfig('./src/UI-Constants/FontSizeWeightFamily.yaml');
+export const DateTimeFormate = await loadConfig('./src/UI-Constants/DateTimeFormate.yaml');
+export const TimeDuration = await loadConfig('./src/UI-Constants/TimeDuration.yaml');
```
Comment (Jha2508) [inline src/Utilities/ConfigHelper.js]: Move all these exports to a new file in Utilites `UIConfig.ts` and export all these , use load config from `ConfigHelper.ts` there

**Review by Jha2508 — DISMISSED** [review]
We only use `ts` and `tsx` files in amaterasu repo, please change accordingly

File: src/Utilities/ConfigHelper.js (hunk 1 above)
Comment (sksinha2410) [reply src/Utilities/ConfigHelper.js]: done

**Comment by Aashirvaad-P** [comment]
@sksinha2410 Pull the changes from develop and resolve the conflicts. Run npm i once and check if everything works right.

File: src/Utilities/ConfigHelper.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import YAML from 'js-yaml';
+import fs from 'react-native-fs'; // npm install react-native-fs
```
Comment (Aashirvaad-P) [inline src/Utilities/ConfigHelper.ts]: remove the comment of npm install

File: src/Utilities/ConfigHelper.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+// Define the type of the configuration object, adjust it based on the actual structure
```
Comment (Aashirvaad-P) [inline src/Utilities/ConfigHelper.ts]: Can this comment be more specific to this usecase

File: src/UI-Constants/Color.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+COLOR_BACKGROUND: '#ecf0f1'
```
Comment (Aashirvaad-P) [inline src/UI-Constants/Color.yaml]: Do we really need so many colors? I think we should keep only those which we are going to use as per the UI designs.

File: src/UI-Constants/Designs.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# Opacities
```
Comment (Aashirvaad-P) [inline src/UI-Constants/Designs.yaml]: This looks good👍🏼

File: src/UI-Constants/Errors.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+NETWORK_ERROR: "Unable to connect. Please check your network connection."
```
Comment (Aashirvaad-P) [inline src/UI-Constants/Errors.yaml]: This one also looks fine. We may modify this as need arises.

File: src/UI-Constants/Color.yaml (hunk 4 above)
Comment (sksinha2410) [reply src/UI-Constants/Color.yaml]: Actually I don't have complete UI design of the app, and is currently being designing. Keeping all this in mind added all this colors that can be used as per need later.

Comment (Aashirvaad-P) [reply src/UI-Constants/Color.yaml]: Ok anyways, we can update it on the go.

File: package.json [hunk 7]
```diff
@@ -14,12 +14,16 @@
...
+    "@types/js-yaml": "^4.0.9",
+    "js-yaml": "^4.1.0",
     "long": "^5.2.3",
     "protobufjs": "^7.3.2",
     "react": "18.2.0",
     "react-native": "0.74.1",
+    "react-native-fs": "^2.20.0",
     "react-native-safe-area-context": "^4.10.4",
     "react-native-screens": "^3.31.1",
+    "yaml": "^2.4.5"
```
Comment (Aashirvaad-P) [inline package.json]: missing comma here

Comment (Aashirvaad-P) [reply package.json]: Always run the project then push the code as already told

Comment (sksinha2410) [reply package.json]: I have run the project sir, that issue must happen by me while solving the conflicts.

Will be taken care from next time.

**Review by Aashirvaad-P — APPROVED** [review]

**Pull request merged**